### PR TITLE
feat: implementation option http2 in `CURLRequest`

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
Added option for version HTTP2 to CURLRequest.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/libraries/curlrequest.rst [hunk 1]
```diff
@@ -332,7 +332,7 @@ is true:
 version
 =======
 
-To set the HTTP protocol to use, you can pass a string or float with the version number (typically either 1.0
-or 1.1, 2.0 is currently unsupported.):
+To set the HTTP protocol to use, you can pass a string or float with the version number (typically either ``1.0``
+or ``1.1``, 2.0 supported since ``v4.3.0``.):
```
Comment (kenjis) [inline user_guide_src/source/libraries/curlrequest.rst:336]:
```suggestion
or ``1.1``, ``2.0`` supported since ``v4.3.0``.):
```

**Comment by kenjis** [comment]
Please add changelog:
https://github.com/codeigniter4/CodeIgniter4/blob/4.3/user_guide_src/source/changelogs/v4.3.0.rst#libraries

File: user_guide_src/source/changelogs/v4.3.0.rst [hunk 2]
```diff
@@ -151,6 +151,7 @@ Model
 Libraries
 =========
 
+- Added option version HTTP2 in ``CURLRequest``
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.3.0.rst:154]:
```suggestion
- Added option version HTTP2 in :ref:`CURLRequest <curlrequest-version>`.
```

File: user_guide_src/source/libraries/curlrequest.rst [hunk 3]
```diff
@@ -332,7 +332,7 @@ is true:
 version
```
Comment (kenjis) [inline user_guide_src/source/libraries/curlrequest.rst:332]:
```suggestion
.. _curlrequest-version:

version
```

File: user_guide_src/source/libraries/curlrequest.rst [hunk 4]
```diff
@@ -332,7 +332,7 @@ is true:
...
 =======
 
-To set the HTTP protocol to use, you can pass a string or float with the version number (typically either 1.0
-or 1.1, 2.0 is currently unsupported.):
+To set the HTTP protocol to use, you can pass a string or float with the version number (typically either ``1.0``
+or ``1.1``, ``2.0`` supported since v4.3.0.):
```
Comment (kenjis) [inline user_guide_src/source/libraries/curlrequest.rst:336]:
```suggestion
or ``1.1``, ``2.0`` is supported since v4.3.0.):
```

**Comment by kenjis** [comment]
```php
<?php

namespace App\Controllers;

class Home extends BaseController
{
    public function index()
    {
        $client   = \Config\Services::curlrequest();
        $response = $client->request(
            'GET',
            'https://api.github.com/user',
            [
                'headers' => [
                    'User-Agent' => 'codeigniter/4.2',
                ],
                'http_errors' => false,
                'version'     => 2.0,
                'debug'       => true,
            ]
        );

        return $response->getStatusCode() . "\n" . $response->getBody();
    }
}
```
```console
$ php public/index.php 
*   Trying 20.27.177.116:443...
* Connected to api.github.com (20.27.177.116) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /usr/local/etc/openssl@1.1/cert.pem
*  CApath: none
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
*  start date: Mar 16 00:00:00 2022 GMT
*  expire date: Mar 16 23:59:59 2023 GMT
*  subjectAltName: host "api.github.com" matched cert's "*.github.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS Hybrid ECC SHA384 2020 CA1
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: GET]
* h2h3 [:path: /user]
* h2h3 [:scheme: https]
* h2h3 [:authority: api.github.com]
* h2h3 [accept: */*]
* h2h3 [user-agent: codeigniter/4.2]
* Using Stream ID: 1 (easy handle 0x7fb7ed026400)
> GET /user HTTP/2
Host: api.github.com
accept: */*
user-agent: codeigniter/4.2

* old SSL session ID is stale, removing
< HTTP/2 401 
< server: GitHub.com
< date: Fri, 14 Oct 2022 01:15:54 GMT
< content-type: application/json; charset=utf-8
< content-length: 131
< x-github-media-type: github.v3; format=json
< access-control-expose-headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
< access-control-allow-origin: *
< strict-transport-security: max-age=31536000; includeSubdomains; preload
< x-frame-options: deny
< x-content-type-options: nosniff
< x-xss-protection: 0
< referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
< content-security-policy: default-src 'none'
< vary: Accept-Encoding, Accept, X-Requested-With
< x-ratelimit-limit: 60
< x-ratelimit-remaining: 57
< x-ratelimit-reset: 1665713617
< x-ratelimit-resource: core
< x-ratelimit-used: 3
< x-github-request-id: F7E1:68C1:2B9C2B:334BE3:6348B84A
< 
* Connection #0 to host api.github.com left intact
401
{"message":"Requires authentication","documentation_url":"https://docs.github.com/rest/reference/users#get-the-authenticated-user"}
```

**Comment by kenjis** [comment]
If I remove the line `'version'     => 2.0,`:
```console
$ php public/index.php 
*   Trying 20.27.177.116:443...
* Connected to api.github.com (20.27.177.116) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /usr/local/etc/openssl@1.1/cert.pem
*  CApath: none
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=GitHub, Inc.; CN=*.github.com
*  start date: Mar 16 00:00:00 2022 GMT
*  expire date: Mar 16 23:59:59 2023 GMT
*  subjectAltName: host "api.github.com" matched cert's "*.github.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS Hybrid ECC SHA384 2020 CA1
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: GET]
* h2h3 [:path: /user]
* h2h3 [:scheme: https]
* h2h3 [:authority: api.github.com]
* h2h3 [accept: */*]
* h2h3 [user-agent: codeigniter/4.2]
* Using Stream ID: 1 (easy handle 0x7fa8b183e800)
> GET /user HTTP/2
Host: api.github.com
accept: */*
user-agent: codeigniter/4.2

* old SSL session ID is stale, removing
< HTTP/2 401 
< server: GitHub.com
< date: Fri, 14 Oct 2022 01:18:00 GMT
< content-type: application/json; charset=utf-8
< content-length: 131
< x-github-media-type: github.v3; format=json
< access-control-expose-headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
< access-control-allow-origin: *
< strict-transport-security: max-age=31536000; includeSubdomains; preload
< x-frame-options: deny
< x-content-type-options: nosniff
< x-xss-protection: 0
< referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
< content-security-policy: default-src 'none'
< vary: Accept-Encoding, Accept, X-Requested-With
< x-ratelimit-limit: 60
< x-ratelimit-remaining: 54
< x-ratelimit-reset: 1665713618
< x-ratelimit-resource: core
< x-ratelimit-used: 6
< x-github-request-id: F844:718C:71DE8D:79B6F0:6348B8C8
< 
* Connection #0 to host api.github.com left intact
401
{"message":"Requires authentication","documentation_url":"https://docs.github.com/rest/reference/users#get-the-authenticated-user"}
```

**Review by kenjis — APPROVED** [review]
LGTM! Thank you.

**Review by MGatner — APPROVED** [review]
Very nice @ddevrs! Thanks for testing @kenjis. Code looks good to me.

**Pull request merged**